source/shared/cpp/ObjectModel/MarkDownParsedResult.cpp

```diff
@@ -0,0 +1,214 @@
+#include "MarkDownParsedResult.h"
+
+using namespace AdaptiveCards;
+
+void MarkDownParsedResult::Translate()
+{
+    MatchLeftAndRightEmphasises();
+}
+
+// appends html block tags at head and tail of the list
+void MarkDownParsedResult::AddBlockTags()
+{
+    // if html list tag, add <ul>, otherwise <p> at the head of the list
+    m_codeGenTokens.front()->MakeItHead();
+
+    // if html list tag, add </ul>, otherwise </p> at the tail of the list
+    m_codeGenTokens.back()->MakeItTail();
+}
+
+// append caller's parsed result to callee's parsed result
+void MarkDownParsedResult::AppendParseResult(MarkDownParsedResult &x)
+{
+    m_codeGenTokens.splice(m_codeGenTokens.end(), x.m_codeGenTokens);
+    m_emphasisLookUpTable.splice(m_emphasisLookUpTable.end(), x.m_emphasisLookUpTable);
+}
+
+// append MarkDownHtmlGenerator object to callee's prased result
+void MarkDownParsedResult::AppendToTokens(const std::shared_ptr<MarkDownHtmlGenerator> &x) 
+{
+    // if two different block types, then add closing tag followed by the opening tag of new type
+    if (!m_codeGenTokens.empty() && 
+        m_codeGenTokens.back()->IsList() != x->IsList())
```

Comment:
>IsList() != x->IsList() [](start = 32, length = 23)

You might consider making changing IsList() to be a function that returns an enum instead a a boolean (GetBlockType()).  That would allow for more block types in the future (ordered vs. unordered lists come to mind)

Reply:
thanks, that's valuable advice

---
In reply to: [157635122](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157635122) [](ancestors = 157635122)